src/core/document.js

```diff
@@ -376,10 +376,7 @@ class Page {
         // is resolved with the complete operator list for a single annotation.
         const opListPromises = [];
         for (const annotation of annotations) {
-          if (
-            isAnnotationRenderable(annotation, intent) &&
-            !annotation.isHidden(annotationStorage)
-          ) {
+          if (isAnnotationRenderable(annotation, annotationStorage, intent)) {
```

Comment:
*Sorry about the back-and-forth here, but all of this looks much more complex than it first seems (with lots of special-cases to consider).*

Previously we did the `isHidden` check *regardless* of the `intent`-value, whereas now it only applies to "display" as far as I can tell, is that intended?

---

I almost wonder if, at this point in time, we should remove the old `isAnnotationRenderable`-helper and simply spell out *exactly* what we want at each call-site instead!?

Reply:
yeah it's intended: a field can be not on screen but printed (from js specs: the link I used to use is dead now)
```
Field is visible on screen and in print               display.visible
Field is hidden on screen and in print              display.hidden
Field is visible on screen but does not print     display.noPrint
Field is hidden on screen but prints                 display.noView
```
And it was likely a bug to not print an hidden field with a flag print.
And yes I think it'd be better to remove `isAnnotationRenderable` helper.

Reply:
> (from js specs: the link I used to use is dead now)

It seems that Adobe recently broke a whole bunch of links, but fortunately the Internet Archive is helpful as usual :-)

The following might be helpful, for future linking purposes, assuming it's the document you're thinking of: https://web.archive.org/web/20201112021418/https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/AcrobatDC_js_api_reference.pdf